hera_mc/cm_health.py

```diff
@@ -0,0 +1,133 @@
+# -*- mode: python; coding: utf-8 -*-
+# Copyright 2018 the HERA Collaboration
+# Licensed under the 2-clause BSD license.
+
+"""Functions to handle various cm table health checks.
+
+"""
+
+from __future__ import absolute_import, print_function
+import warnings
+from hera_mc import mc, cm_utils
+from hera_mc import part_connect as PC
+from hera_mc import cm_revisions
+
+
+class RevisionError(Exception):
+    def __init__(self, hpn):
+        # Call the base class constructor with the parameters it needs
+        message = "Multiple revisions found on {}".format(hpn)
+        super(RevisionError, self).__init__(message)
+
+
+class Connections:
+    def __init__(self, session=None):
+        if session is None:
+            db = mc.connect_to_mc_db(None)
+            self.session = db.sessionmaker()
+        else:
+            self.session = session
+        self.conndict = None
+
+    def get_conndict(self, session=None):
+        self.conndict = {}
+        self.multiples = []
+        self.num_connections = 0
+        for conn in self.session.query(PC.Connections).all():
+            self.num_connections += 1
+            connection = [conn.upstream_part, conn.up_part_rev, conn.upstream_output_port,
+                          conn.downstream_part, conn.down_part_rev, conn.downstream_input_port]
+            connection = [x.lower() for x in connection]
+            k = ':'.join(connection)
+            if k in self.conndict.keys():
+                self.conndict[k].append([conn.start_gpstime, conn.stop_gpstime])
```

Comment:
Should these GPS times be converted to Astropy times at this point? It looks like most if not all of the code below converts the times to Astropy times whenever they're used, so it might be better to do the conversions up-front and not have to worry about them later.